node/silkworm/etl/buffer.cpp

```diff
@@ -22,25 +22,17 @@ namespace silkworm::etl {
 
 void Buffer::put(const Entry& entry) {
     size_ += entry.size();
-
-    assert(length_ <= buffer_.size());
-    if (length_ == buffer_.size()) {
-        buffer_.push_back(entry);
-    } else {
-        buffer_[length_] = entry;
-    }
-
-    ++length_;
+    buffer_.push_back(entry);
 }
 
-void Buffer::sort() { std::sort(buffer_.data(), buffer_.data() + length_); }
+void Buffer::sort() { std::sort(buffer_.begin(), buffer_.end()); }
 
 size_t Buffer::size() const noexcept { return size_; }
 
-gsl::span<const Entry> Buffer::entries() const noexcept { return {buffer_.data(), length_}; }
+    gsl::span<const Entry> Buffer::entries() const noexcept { return {buffer_.data(), buffer_.size()}; }
```

Comment:
I believe with this mod the gsl::span is not needed and we could return a direct const reference to buffer_ to iterate the items.

Reply:
> I believe with this mod the gsl::span is not needed and we could return a direct const reference to buffer_ to iterate the items.

True, but `gsl::span` is still valid, so it is just a matter of preference. I am happy with either. In C++20 span is in the standard (`std::span`).

Reply:
As here we don't return a subset of items returning a non-owning container with the same set of items of the original container does not make any sense semantically. Imho

Reply:
Yes I agree with @AndreaLanfranchi , since we mean to return the whole vector it is just more natural (and more efficient) to return a reference to the vector itself, rather than calculating the vector's length and constructing a new object. I have checked in the change.